CMake: Add option to download Qt binaries

yuriks opened the pull request:
This adds an option in CMake that'll download a binary package of Qt binaries. With this (and the #1019 workaround also included in this branch) it's possible to compile Citra with VS2015.

jroweboy commented on CMakeLists.txt:
```diff
@@ -77,7 +77,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/externals/cmake-modules")
 include_directories(${OPENGL_INCLUDE_DIR})
 
-option(ENABLE_GLFW "Enable the GLFW frontend" ON)
+option(ENABLE_GLFW "Enable the GLFW frontend" OFF)
```
Does the GLFW build just not work on VS2015? Just wondering why you are disabling it by default.

yuriks replied:
It also needs a separate set of binaries, and I haven't built those yet.

bunnei commented on CMakeLists.txt:
```diff
@@ -153,12 +153,31 @@ ELSE()
+        if (NOT EXISTS "${QTPATH}")
+            message(STATUS "Downloading Qt binaries for ${QTVER}...")
+            file(DOWNLOAD
+                https://github.com/yuriks/citra-depends/raw/master/${QTVER}.7z
```
Submodule? I've kind of wanted to do this for a while anyway to alleviate the burden of Windows users having to download/install a 600MB binary. FWIW, also, what's the point of having this be nested zips?

yuriks replied:
Nested zips?

Not a submodule so that people don't need to download all binaries for all platforms ever. Just the one they're going to use.

neobrain replied:
If this stays as is, the repo should be hosted in the citra-emu organization.

yuriks commented:
Btw, if we go forward with this approach, then I plan on using the same system for GLFW or SDL2 binaries. (This decision is in fact blocking #841 right now.)

Subv commented:
What's the status of this RFC? I've been using the patch for a while now and it seems to be the best way to support VS2015 without having to wait for Qt to update their binaries.

yuriks commented:
@Subv: Was just waiting for more of a reaction. Seems like we agree to go with this, so I'll clean up the code and add a GLFW build.

yuriks commented:
Updated.
  - Added GLFW libraries for 2015 and 2013
  - Extracted downloading and extracting code to a CMake function
  - Fixed architecture detection in CMake (will split up in another commit later)

yuriks commented:
I think this is pretty much done. Just need to review and then create an official repo for the files before merging.

yuriks commented:
Updated fixing CI failures.

bunnei commented:
Finally got around to testing this after struggling a bit with my environment. Everything checks out fine here. Let's move forward with setting up the repo for the binaries in the Citra org.

yuriks commented:
Updated. Let's just check that the AppVeyor build is working fine.

bunnei commented:
LGTM
